src/road_cmd.cpp

```diff
@@ -2429,7 +2429,7 @@ CommandCost CmdConvertRoad(TileIndex tile, DoCommandFlag flags, uint32 p1, uint3
 
 			if (flags & DC_EXEC) { // we can safely convert, too
 				/* Update the company infrastructure counters. */
-				if (owner == _current_company) {
+				if (owner == _current_company && Company::IsValidID(_current_company)) {
 					Company * c = Company::Get(_current_company);
 					c->infrastructure.road[from_type] -= num_pieces;
 					c->infrastructure.road[to_type] += num_pieces;
```

Comment:
Based on my reading of https://github.com/OpenTTD/OpenTTD/commit/0125892f041026d049dd9ce7e6d1a63ed5fd67eb, but not tested, I think
```suggestion
				if (owner == _current_company) {
					ConvertRoadTypeOwner(tile, num_pieces, owner, from_type, to_type);
```
should be a valid option too.

Reply:
Yes, that works fine, cleans up some duplication of code and makes sure the infrastructure window is properly dirtied when converting road.